libs/providers/growthbook-client/src/lib/translate-result.ts

```diff
@@ -0,0 +1,29 @@
+import { FeatureResult } from '@growthbook/growthbook';
+import { ErrorCode, ResolutionDetails } from '@openfeature/web-sdk';
+
+const FEATURE_RESULT_ERRORS = ['unknownFeature', 'cyclicPrerequisite'];
+
+function translateError(errorKind?: string): ErrorCode {
+  switch (errorKind) {
+    case 'unknownFeature':
+      return ErrorCode.FLAG_NOT_FOUND;
+    case 'cyclicPrerequisite':
+      return ErrorCode.PARSE_ERROR;
+    default:
+      return ErrorCode.GENERAL;
+  }
+}
+
+export default function translateResult<T>(result: FeatureResult, defaultValue: T): ResolutionDetails<T> {
+  const resolution: ResolutionDetails<T> = {
+    value: result.value === null ? defaultValue : result.value,
+    reason: result.source,
+    variant: result.experimentResult?.key,
+  };
+
+  if (FEATURE_RESULT_ERRORS.includes(result.source)) {
+    resolution.errorCode = translateError(result.source);
+  }
+
+  return resolution;
+}
```

Comment:
This function takes a `FeatureResult` as argument which is a `FeatureResult<any>`, so `result.value` is any, which means the types returned by this function are not save.

I added the typical check in the suggested change.
Another option would be to try to convert the types, but the common way is to throw a `TypeMismatchError` which will  result in the default value being returned by the client.

If you accept the change, please add a small test case @msamper.

```suggestion
export default function translateResult<T>(result: FeatureResult<unknown>, defaultValue: T): ResolutionDetails<T> {
  if (result.value !== null && typeof result.value !== typeof defaultValue) {
    throw new TypeMismatchError(`Expected flag type ${typeof defaultValue} but got ${typeof result.value}`);
  }

  const resolution: ResolutionDetails<T> = {
    value: result.value === null ? defaultValue : (result.value as T),
    reason: result.source,
    variant: result.experimentResult?.key,
  };

  if (FEATURE_RESULT_ERRORS.includes(result.source)) {
    resolution.errorCode = translateError(result.source);
  }

  return resolution;
}
```